hudi-common/src/main/java/org/apache/hudi/BaseHoodieTableFileIndex.java

```diff
@@ -229,17 +238,93 @@ protected List<FileSlice> getInputFileSlices(PartitionPath partition) {
   }
 
   private Map<PartitionPath, List<FileSlice>> loadFileSlicesForPartitions(List<PartitionPath> partitions) {
-    FileStatus[] allFiles = listPartitionPathFiles(partitions);
+    Pair<Map<Path, FileStatus[]>, Map<String, FileStatus[]>> partitionFilesPair = listPartitionPathFiles(partitions);
     HoodieTimeline activeTimeline = getActiveTimeline();
     Option<HoodieInstant> latestInstant = activeTimeline.lastInstant();
 
-    HoodieTableFileSystemView fileSystemView =
-        new HoodieTableFileSystemView(metaClient, activeTimeline, allFiles);
-
     Option<String> queryInstant = specifiedQueryInstant.or(() -> latestInstant.map(HoodieInstant::getTimestamp));
 
     validate(activeTimeline, queryInstant);
 
+    int parallelism = Integer.parseInt(String.valueOf(configProperties.getOrDefault(HoodieCommonConfig.TABLE_LOADING_PARALLELISM.key(),
+        HoodieCommonConfig.TABLE_LOADING_PARALLELISM.defaultValue())));
+
+    Map<PartitionPath, List<FileSlice>> cachedAllInputFileSlices;
+    long buildCacheFileSlicesLocalStart = System.currentTimeMillis();
+    if (parallelism > 0 && partitions.size() > 0) {
+
+      // convert Map<Path, FileStatus[]> to Map<String, FileStatus[]>
+      Map<String, FileStatus[]> left = partitionFilesPair.getLeft().entrySet().stream().map(entry -> {
```

Comment:
Wouldn't it be simpler to have return type of listPartitionPathFiles use String instead of Path for partitionPath. We can avoid these transformations.